testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/jca/connectionlistener/AbstractTestsuite.java

```diff
@@ -0,0 +1,301 @@
+            operation.get("user-name").set("sa");
+            operation.get("password").set("sa");
+            
+            
+            operation.get("connection-listener-class").set(CONNECTION_LISTENER_CLASS_IMPL);
+            executeOperation(operation);
+                   
+            if (xa) {
+                final ModelNode xaDatasourcePropertiesAddress = address.clone();
+                xaDatasourcePropertiesAddress.add("xa-datasource-properties", "URL");
+                xaDatasourcePropertiesAddress.protect();
+                final ModelNode xaDatasourcePropertyOperation = new ModelNode();
+                xaDatasourcePropertyOperation.get(OP).set("add");
+                xaDatasourcePropertyOperation.get(OP_ADDR).set(xaDatasourcePropertiesAddress);
+                xaDatasourcePropertyOperation.get("value").set("jdbc:h2:mem:test");
+
+                executeOperation(xaDatasourcePropertyOperation);
+            }
+
+            operation = new ModelNode();
+            operation.get(OP).set("enable");
+            operation.get(OP_ADDR).set(address);
+
+            executeOperation(operation);
+
+            return address;
+        }
+    }
+    
+    protected void modifyModuleIronJacamarJdbc() throws Exception{
+    	addModule(defaultPath);
+    	createSubModuleJarConnectionListenerImpl(JAR_NAME);
+    	//we have to edit module.xml and add our new JAR as resource
+    	
+    	File moduleXml = new File(testModuleRoot, MODULE_XML_NAME);
+    	if(moduleXml.exists() && moduleXml.isFile()){
+    		File backupModuleXml = new File(testModuleRoot, MODULE_XML_BACKUP_NAME);
+    		copyFile(backupModuleXml, new FileInputStream(moduleXml));
+    		
+    		try(BufferedReader in = new BufferedReader(new InputStreamReader(new FileInputStream(backupModuleXml)));
+    				PrintWriter out = new PrintWriter(new File(testModuleRoot, MODULE_XML_NAME));) {
+    			String line;
+    			while ((line = in.readLine()) != null) {
+    				out.println(line);
+    				
+    				if(line.trim().startsWith("<resource-root")){
+    					out.println("<resource-root path=\""+JAR_NAME+"\"/>");
+    				}
+    			}
+    		}
+    	}else{
+    		throw new IllegalStateException(MODULE_XML_NAME + " must exists");
+    	}
+    	
+    	
+    	
+    }    
+    
+    public void restoreModuleXml(){
+    	try{
+    	File moduleBackupXml = new File(testModuleRoot, MODULE_XML_BACKUP_NAME);
+    	if(moduleBackupXml.exists() && moduleBackupXml.isFile()){
+    		File moduleXml = new File(testModuleRoot, MODULE_XML_NAME);
+    		moduleXml.delete();
+    		moduleBackupXml.renameTo(new File(testModuleRoot, MODULE_XML_NAME));
+    	}
+    	}catch(Exception e){
+    		log.warn(e.getMessage());
+    	}
+    	
+    	try {
+    		if(moduleJar != null){
+    			moduleJar.delete();
+    		}
+		} catch (Exception e) {
+			log.warn(e.getMessage());
+		}
+    }
+
+    public void addModule(final String moduleName) throws Exception {
```

Comment:
why so much duplication of test code? this is the 4th (5th) time jca tests add the same code, and we need to fix it on 5 places if(when) it gets broken.
Take a look at AbstractModuleDeploymentTestCaseSetup which does pretty much all this already. and is also only used for JCA tests